src/renderers/dom/shared/inputValueTracking.js

```diff
@@ -109,70 +108,61 @@ function trackValueOnNode(node: any, inst: any): ?ValueTracker {
 
 var inputValueTracking = {
   // exposed for testing
-  _getTrackerFromNode(node: ElementWithWrapperState) {
-    return getTracker(ReactDOMComponentTree.getInstanceFromNode(node));
+  _getTrackerFromNode(inst: ElementWithValueTracker) {
+    return getTracker(inst);
   },
 
-  trackNode(node: ElementWithWrapperState) {
+  _getTrackerFromInstance(inst: ReactInstance | Fiber) {
+    return getTracker(ReactDOMComponentTree.getNodeFromInstance(inst));
+  },
+
+  trackNode(node: ElementWithValueTracker) {
     if (getTracker(node)) {
       return;
     }
-    node._wrapperState.valueTracker = trackValueOnNode(node, node);
+
+    // TODO: Once it's just Fiber we can move this to node._wrapperState
+    node._valueTracker = trackValueOnNode(node);
   },
 
-  track(inst: InstanceWithWrapperState) {
-    if (getTracker(inst)) {
-      return;
-    }
-    var node = ReactDOMComponentTree.getNodeFromInstance(inst);
-    inst._wrapperState.valueTracker = trackValueOnNode(node, inst);
+  track(inst: ReactInstance | Fiber) {
+    inputValueTracking.trackNode(
+      ReactDOMComponentTree.getNodeFromInstance(inst),
+    );
   },
 
-  // TODO: in practice "subject" can currently be Stack instance,
-  // Fiber, or DOM node. This is hard to understand. We should either
-  // make it accept only DOM nodes, or only Fiber/Stack instances.
-  updateValueIfChanged(subject: SubjectWithWrapperState | Fiber) {
-    if (!subject) {
+  updateNodeValueIfChanged(node: ElementWithValueTracker) {
+    if (!node) {
       return false;
     }
 
-    var isNode = (subject: any).nodeType === ELEMENT_NODE;
-    var tracker = getTracker(subject);
+    var tracker = getTracker(node);
+    // if there is no tracker at this point it's unlikely
+    // that trying again will succeed
     if (!tracker) {
-      if (isNode) {
-        // DOM node
-        inputValueTracking.trackNode((subject: any));
-      } else if (typeof (subject: any).tag === 'number') {
-        // Fiber
-        inputValueTracking.trackNode((subject: any).stateNode);
-      } else {
-        // Stack
-        inputValueTracking.track((subject: any));
-      }
       return true;
     }
 
-    var node;
-    if (isNode) {
-      // DOM node
-      node = subject;
-    } else {
-      // Fiber and Stack
-      node = ReactDOMComponentTree.getNodeFromInstance(subject);
-    }
-
     var lastValue = tracker.getValue();
     var nextValue = getValueFromNode(node);
     if (nextValue !== lastValue) {
       tracker.setValue(nextValue);
       return true;
     }
-
     return false;
   },
 
-  stopTracking(inst: InstanceWithWrapperState | Fiber) {
-    var tracker = getTracker(inst);
+  updateValueIfChanged(instOrFiber: ReactInstance | Fiber) {
```

Comment:
That's fine with me, it just means we need to call getNode at the call sites: ReactDOMComponent and ChangeEventPlugin